datafusion/core/tests/sqllogictests/src/setup.rs

```diff
@@ -39,6 +41,95 @@ pub async fn register_aggregate_tables(ctx: &SessionContext) {
     register_test_data(ctx);
 }
 
+pub async fn register_group_tables(ctx: &SessionContext) {
+    register_time_test_tables(ctx);
+}
+
+/// Create and populate time32_s, time32_ms, time64_us, time64_ns tables
+fn register_time_test_tables(ctx: &SessionContext) {
```

Comment:
I don't think you need rust code to do this setup anymore

You can create arbitrary Arrow types that have no corresponding SQL types with the `arrow_cast` function -- https://arrow.apache.org/datafusion/user-guide/sql/data_types.html

For example, here is one way to do it:
https://github.com/apache/arrow-datafusion/blob/7759d96e892e7c159136af35ebc0def49ed3b3b1/datafusion/core/tests/sqllogictests/test_files/timestamps.slt#L113-L129

So you can make a value of `Time32(Seconds)` with something like

```sql
❯ select arrow_cast(5000::int, 'Time32(Second)');
+-------------+
| Int64(5000) |
+-------------+
| 01:23:20    |
+-------------+
```

Reply:
Alright, thank you for the feedback @alamb. Didn't know that. I'm done for the day, will probably work more on this tomorrow if nothing comes up.